### DISCUSSION TITLE
Outbound attribute coercion

### DISCUSSION THREAD

**Discussion by albertosaez** [dcm4che/dcm4chee-arc-light#2941] [Q&A] [answered]
Is it possible to configure a "dynamic" coercion of attributes for  for some destinations , based on a REST query ?

It would be similar to the existing functionality of "[leading archive coercion](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Proxy-query-requests-to-another-Archive#leading-archive-attribute-coercion)",  what is able to coerce attributes based on a C-FIND query. It has a lot of sense for coercing from other PACS ,but it's not dev-friendly for other scenarios.

Use-case description.

In a referral scenario, the company that is the destination of the referral has to send to the hospital source of the referral, the image studies, but using their identifications (patientID, acc. number, ....) . 
In order to do this coercion, we would need to change these attributes based on a table of equivalence.
Is it possible to do this coercion with dcm4chee-arc ?  Using the "leading archive coercion" could be possible, but then it would be needed to create a C-find server to provide these relationship. Is there any other alternative ?

**Comment by gunterze** [comment, marked as answer]
Should become possible with
- C-STORE SCU: Optionally nullify Accession Number in retrieved/exported Composite Objects dependent on Issuer of Accession Number #2958
- C-STORE SCU: Optionally adjust/nullify Patient ID in retrieved/exported Composite Objects dependent on Issuer of Patient ID #2962

as sub-tasks of
- Support IHE SWF Scheduled Workflow Multiple Identity Resolution option #2964

You may then plug-in your own implementation of an [Patient Identifier Cross-Reference (PIX) Consumer Service Provider](https://github.com/dcm4che/dcm4chee-arc-light/issues/2955) querying your Patient ID cross-reference back-end, invoked by #2962 to find corresponding Patient ID of the Destination Assigning Authority.

I am not aware of any use case of cross-referencing Accessing Numbers of different Assigning Authorities.

**Comment by albertosaez** [comment]
Well, definitely, the[ J.2.7: ](https://ihe.net/Technical_Framework/upload/IHE_RAD_Suppl_MIMA.pdf#page=118) from MIMA cover the requirements, but as you mentioned, I don't see the way of getting the cross-accessingNumber in a similar way as in the PIX profile. And sometimesm it's not need only to change the acc.number + patientID, but also other attributes. For this the "leading archive coercion" functionality is superior (but requires a c-find SCP).

I've identified any IHE profile for this use case ,what is strange, as it's the way all NHS works with other companies when referring patients.

**Comment by gunterze** [comment]
Opened https://github.com/dcm4che/dcm4chee-arc-light/issues/2967 .
Recommend to post use cases not (yet) addressed by any IHE Integration profile to the 
IHE Radiology Planning Committee (https://groups.google.com/g/ihe-rad-plan ? - I currently can't access that address) or/and
IHE Radiology Technical Committee (https://groups.google.com/g/ihe-rad-tech ).
